src/requests.rs

```diff
@@ -597,6 +600,69 @@ impl TryFrom<&OriginalUploadSigningKeysRequest> for UploadSigningKeysRequest {
     }
 }
 
+#[wasm_bindgen(getter_with_clone)]
+#[derive(Debug)]
+/// Upload a dehydrated device to the server.
+pub struct PutDehydratedDeviceRequest {
+    /// A JSON-encoded string containing the rest of the payload: `rooms`.
+    ///
+    /// It represents the body of the HTTP request.
+    #[wasm_bindgen(readonly)]
+    pub body: JsString,
+}
+
+#[wasm_bindgen]
+impl PutDehydratedDeviceRequest {
+    /// Create a new `PutDehydratedDeviceRequest`
+    #[wasm_bindgen(constructor)]
+    pub fn new(body: JsString) -> PutDehydratedDeviceRequest {
+        Self { body }
+    }
+}
+
+impl TryFrom<OriginalPutDehydratedDeviceRequest> for PutDehydratedDeviceRequest {
```

Comment:
I think there is a macro to do that.
Like `request!(KeysBackupRequest from OriginalKeysBackupRequest extracts version: string and groups rooms);`

Reply:
I think that the macro only works for requests that have an ID (which is why `UploadSigningKeys` is also not done using the macro).

Reply:
It feels weird to me that this has to happen in the bindings, macro or not. Surely turning the request object into the relevant JSON is something that all applications will need to do, so it should be in the main SDK?

I appreciate it's following a pattern, so this is maybe something to fix another time, but still, it seems wrong.